src/variants/types/methylation.rs

```diff
@@ -0,0 +1,473 @@
+                    0
+                };
+
+                matches!(
+                    self.locus().overlap(left, false, start_offset_left, 0),
+                    Overlap::Enclosing
+                ) || matches!(
+                    self.locus().overlap(right, false, start_offset_right, 0),
+                    Overlap::Enclosing
+                )
+            }
+        } {
+            if match self.readtype {
+                // Some single PacBio reads don't have an MM:Z value and are therefore not legal
+                Readtype::Illumina => true,
+                Readtype::PacBio => mm_exists(evidence),
+                Readtype::Nanopore => mm_exists(evidence),
+            } {
+                Some(vec![0])
+            } else {
+                None
+            }
+        } else {
+            None
+        }
+    }
+
+    fn allele_support(
+        &self,
+        evidence: &Evidence,
+        _alignment_properties: &AlignmentProperties,
+        _alt_variants: &[Box<dyn Realignable>],
+    ) -> Result<Option<AlleleSupport>> {
+        match evidence {
+            Evidence::SingleEndSequencingRead(read) => match self.readtype {
+                Readtype::Illumina => Ok(self.allele_support_per_read(read, false)?),
+                Readtype::PacBio | Readtype::Nanopore => {
+                    Ok(self.allele_support_per_read(read, true)?)
+                }
+            },
+            Evidence::PairedEndSequencingRead { left, right } => match self.readtype {
+                Readtype::Illumina => {
+                    let left_support = self.allele_support_per_read(left, false)?;
+                    let right_support = self.allele_support_per_read(right, false)?;
+
+                    match (left_support, right_support) {
+                        (Some(mut left_support), Some(right_support)) => {
+                            left_support.merge(&right_support);
+                            Ok(Some(left_support))
+                        }
+                        (Some(left_support), None) => Ok(Some(left_support)),
+                        (None, Some(right_support)) => Ok(Some(right_support)),
+                        (None, None) => Ok(None),
+                    }
+                }
+                Readtype::PacBio | Readtype::Nanopore => {
+                    let left_support = self.allele_support_per_read(left, true)?;
+                    let right_support = self.allele_support_per_read(right, true)?;
+
+                    match (left_support, right_support) {
+                        (Some(mut left_support), Some(right_support)) => {
+                            left_support.merge(&right_support);
+                            Ok(Some(left_support))
+                        }
+                        (Some(left_support), None) => Ok(Some(left_support)),
+                        (None, Some(right_support)) => Ok(Some(right_support)),
+                        (None, None) => Ok(None),
+                    }
```

Comment:
I am sure this can be simplified by doing the type check inside the argument passing to allele_support_per_read. Lot of redundancy here.